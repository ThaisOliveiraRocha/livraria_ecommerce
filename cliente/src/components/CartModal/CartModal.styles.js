import styled, { keyframes } from "styled-components";

const fade = keyframes`
  from {
        opacity: 0;
        transform: scale(0.9);
    }
    to {
        opacity: 1;
        transform: scale(1);
    }
`;

export const Body = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  box-sizing: content-box;
  position: absolute;
  top: 60px;
  right: 10px;

  width: 300px;
  max-height: 400px;
  min-height: 60px;
  z-index: 10;
  padding: 10px;
  border: 1px solid #79bac1;
  box-shadow: 5px 5px 30px rgba(121, 186, 193, 0.5);
  background-color: white;

  animation: ${fade} 250ms;
  animation-fill-mode: backwards;
`;

export const ModalComponent = styled.div`
  max-height: 300px;
  overflow-y: auto;
`;

export const TituloContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  top: 20px;
  height: 30px;
  width: 100%;
`;

export const Table = styled.table`
  font-size: 16px;
  font-weight: 600;
  border-collapse: collapse;
  margin-top: 15px;
  thead{
    border-bottom: 1px solid gray;
  }
`;

export const Td = styled.td`
  width: 70px;
  font-weight: normal;
`;

export const Th = styled.th`
  width: 70px;
  font-weight: normal;
`;

export const TdFooter = styled.td`
  display: flex;
  flex-grow: 1;
  font-weight: normal;
`;

export const ImageBook = styled.img`
  width: 40px;
  height: auto;

  margin: 10px;
`;

export const InfoBook = styled.span`
  font-size: 14px;
  margin-right: 5px;
`;

export const ImageTrash = styled.img`
  width: 20px;
  height: auto;
  cursor: pointer;

  &:active {
    transform: scale(0.9, 0.9);
  }
`;

export const TotalContainer = styled.div`
  margin-top: 10px;
  margin-bottom: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const ButtonGroup = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const Overlay = styled.div`
  position: fixed;
  /* display: none; */
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 2;
`;

export const Linha = styled.div`
  border-top: 1px solid rgba(11, 107, 230, 0.5);
`;
