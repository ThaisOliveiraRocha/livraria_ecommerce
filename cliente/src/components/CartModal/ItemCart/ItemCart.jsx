import React, { useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Tr, Td, ImageBook, ImageTrash } from "./ItemCart.styles";
import Text from "../../Text";
import { showModal } from "../../../store/actions";
import { withRouter } from "react-router-dom";

export const ItemCart = ({ item, deleteItem }) => {
  return (
    <>
      <Tr>
        <Td>
          <ImageBook src={item.imagem} />
        </Td>
        <Td>
          <Text>{item.titulo}</Text>
        </Td>
        <Td>
          <Text>{item.qtd}</Text>
        </Td>
        <Td>
          <Text>R${item.preco}</Text>
        </Td>
        <Td>
          <ImageTrash
            className="btnLixeira"
            src={require("../../../assets/images/excluir.png")}
            onClick={() => deleteItem(item.titulo)}
          />
        </Td>
      </Tr>
      
    </>
  );
};

ItemCart.propTypes = {
  // bla: PropTypes.string,
};
const mapStateToProps = state => ({
  // item: state.livraria.carrinho
});

const mapDispatchToProps = dispatch => ({
  deleteItem: titulo => {
    dispatch(showModal(titulo, true));
  }
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ItemCart));
