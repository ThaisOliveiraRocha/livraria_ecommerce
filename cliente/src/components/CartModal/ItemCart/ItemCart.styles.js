import styled from "styled-components";

export const Tr = styled.tr`
  /* display: flex;
  align-items: center;
  width: 100%; */
`;

export const Td = styled.td`
  text-align: center;
  width: 70px;
  font-weight: normal;
`;

export const ImageBook = styled.img`
  width: 40px;
  height: auto;

  margin: 10px;
`;

export const InfoBook = styled.span`
  font-size: 14px;
  margin-right: 5px;
`;

export const Input = styled.input`
  font-size: 14px;
  width: 50px;
  margin-right: 5px;
`;

export const ImageTrash = styled.img`
  width: 20px;
  height: auto;
  cursor: pointer;

  &:active {
    transform: scale(0.9, 0.9);
  }
`;
