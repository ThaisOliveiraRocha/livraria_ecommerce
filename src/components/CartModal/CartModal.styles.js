import styled, { keyframes } from "styled-components";

const fade = keyframes`
  from {
        opacity: 0;
        transform: scale(0.9);
    }
    to {
        opacity: 1;
        transform: scale(1);
    }
`;

export const ModalBody = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  box-sizing: content-box;
  position: absolute;
  top: 65px;
  width: 300px;
  right: 10px;

  border: 1px solid rgba(11, 107, 230, 0.5);
  box-shadow: 5px 5px 30px rgba(11, 107, 230, 0.3);
  background-color: white;

  /* height: 170px; */
  overflow-y: scroll;
  z-index: 10;

  padding: 10px;

  animation: ${fade} 250ms;
  animation-fill-mode: backwards;
`;

export const Table = styled.table`
  font-size: 16px;
  font-weight: 600;
  border-collapse: collapse;
  margin-top: 15px;

  thead{
    border-bottom: 1px solid gray;
  }
`;

export const Titulo = styled.div`
  font-size: 16px;
  font-weight: 600;
`;

export const Tr = styled.tr`
  /* display: flex;
  align-items: center;
  width: 100%; */
`;

export const Td = styled.td`
  width: 70px;
  font-weight: normal;
`;

export const Th = styled.th`
  width: 70px;
  font-weight: normal;
`;

export const TdFooter = styled.td`
  display: flex;
  flex-grow: 1;
  font-weight: normal;
`;

export const ImageBook = styled.img`
  width: 40px;
  height: auto;

  margin: 10px;
`;

export const InfoBook = styled.span`
  font-size: 14px;
  margin-right: 5px;
`;

export const Input = styled.input`
  font-size: 14px;
  width: 50px;
  margin-right: 5px;
`;

export const ImageTrash = styled.img`
  width: 20px;
  height: auto;
  cursor: pointer;

  &:active {
    transform: scale(0.9, 0.9);
  }
`;

export const TotalContainer = styled.div`
  margin-bottom: 10px;
`;

export const ButtonGroup = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const Overlay = styled.div`
  position: fixed;
  /* display: none; */
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 2;
`;

export const Linha = styled.div`
  border-top: 1px solid rgba(11, 107, 230, 0.5);
`;
