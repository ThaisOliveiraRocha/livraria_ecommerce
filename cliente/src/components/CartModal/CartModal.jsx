import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  Body,
  ModalComponent,
  TituloContainer,
  Table,
  Th,
  TotalContainer,
  ButtonGroup,
  Overlay
} from "./CartModal.styles";
import Titulo from "../Title";
import Text from "../Text";
import Button from "../Button";
import ItemCart from "./ItemCart/ItemCart";
import { withRouter } from "react-router-dom";
import { getCarrinho } from "../../store/actions";

const somaCarrinho = carrinho => {
  let soma = 0;
  carrinho.map((item, index) => {
    soma += item.qtd * item.preco;
  });
  return soma.toFixed(2);
};

export const CartModal = ({ carrinho, showCarrinho, onClose }) => {
  if (carrinho.length === 0) {
    return (
      <>
        <Overlay onClick={onClose} />
        <Body>
          <Titulo>Meu Carrinho</Titulo>
          <Text>Carrinho vazio! Comece a comprar agora.</Text>
        </Body>
      </>
    );
  }
  return (
    <>
      <Overlay onClick={onClose} />
      <Body>
        <TituloContainer>
          <Titulo>Meu carrinho</Titulo>
        </TituloContainer>
        <ModalComponent>
          <Table>
            <thead>
              <tr>
                <Th></Th>
                <Th>
                  <Text>Título</Text>
                </Th>
                <Th>
                  <Text>Qnt.</Text>
                </Th>
                <Th>
                  <Text>Preço</Text>
                </Th>
                <Th></Th>
              </tr>
            </thead>
            <tbody>
              {carrinho.map((item, index) => {
                return <ItemCart key={index} item={item} />;
              })}
            </tbody>
          </Table>
        </ModalComponent>
        <TotalContainer>
          <Titulo>Total da compra:</Titulo>
          <Text> R$ {somaCarrinho(carrinho)}</Text>
        </TotalContainer>
        <ButtonGroup>
          <Button className="btnComprar" onClick={() => showCarrinho(carrinho)}>
            <Text color="white">Finalizar Compra</Text>
          </Button>
        </ButtonGroup>
      </Body>
    </>
  );
};

CartModal.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  carrinho: state.livraria.carrinho
});

const mapDispatchToProps = (dispatch, { history }) => ({
  showCarrinho: carrinho => {
    dispatch(getCarrinho(carrinho));
    history.push("/carrinho");
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(CartModal)
);
