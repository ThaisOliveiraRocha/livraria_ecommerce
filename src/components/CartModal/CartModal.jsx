import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  ModalBody,
  Table,
  Tr,
  Th,
  TotalContainer,
  ButtonGroup,
  Overlay
} from "./CartModal.styles";
import Titulo from "../../components/Title";
import Text from "../../components/Text";
import Button from "../../components/Button";
import ItemCart from "./ItemCart/ItemCart";
import { withRouter } from "react-router-dom";
import { getCarrinho } from "../../store/actions";

const somaCarrinho = carrinho => {
  let soma = 0;
  carrinho.map((item, index) => {
    soma += item.qtd * item.preco;
  });
  return soma.toFixed(2);
};

const CartModal = ({ carrinho, showCarrinho, onClose }) => {
  if (carrinho.length === 0) {
    return (
      <>
      <Overlay onClick={onClose}/>
      <ModalBody>
        <Titulo>Meu Carrinho</Titulo>
        <Text>Carrinho vazio! Comece a comprar agora.</Text>
      </ModalBody>
      </>
    );
  }
  return (
    <>
      <Overlay onClick={onClose}/>
      <ModalBody>
        <Titulo>Meu carrinho</Titulo>
        <Table>
          <thead>
            <Tr>
              <Th></Th>
              <Th><Text>Título</Text></Th>
              <Th><Text>Qnt.</Text></Th>
              <Th><Text>Preço</Text></Th>
              <Th></Th>
            </Tr>
          </thead>
          <tbody>
            {carrinho.map((item, index) => {
              return <ItemCart key={index} item={item} />;
            })}
          </tbody>
        </Table>
        <TotalContainer>
          <Titulo>Total da compra:</Titulo>
          <Text> R$ {somaCarrinho(carrinho)}</Text>
        </TotalContainer>
        <ButtonGroup>
          <Button onClick={() => showCarrinho(carrinho)}>
            Finalizar Compra
          </Button>
        </ButtonGroup>
      </ModalBody>
    </>
  );
};

CartModal.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  carrinho: state.livraria.carrinho
});

const mapDispatchToProps = (dispatch, { history }) => ({
  showCarrinho: carrinho => {
    dispatch(getCarrinho(carrinho));
    history.push("/carrinho");
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(CartModal)
);
