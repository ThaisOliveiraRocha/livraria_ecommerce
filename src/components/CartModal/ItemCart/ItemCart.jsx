import React, { useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Tr, Td, ImageBook, ImageTrash } from "./ItemCart.styles";
import Text from "../../../components/Text";
import { showModal } from "../../../store/actions";
import RemoveItemModal from "../../RemoveItemModal";

const ItemCart = ({ item, deleteItem }) => {
  return (
    <>
      <Tr>
        <Td>
          <ImageBook src={require(`../../../assets/images/${item.imagem}`)} />
        </Td>
        <Td>
          <Text>{item.titulo}</Text>
        </Td>
        <Td>
          <Text>{item.qtd}</Text>
        </Td>
        <Td>
          <Text>R${item.preco}</Text>
        </Td>
        <Td>
          <ImageTrash
            src={require("../../../assets/images/excluir.png")}
            onClick={() => deleteItem(item.titulo)}
          />
        </Td>
      </Tr>
      
    </>
  );
};

ItemCart.propTypes = {
  // bla: PropTypes.string,
};
const mapStateToProps = state => ({
  // item: state.livraria.carrinho
});

const mapDispatchToProps = dispatch => ({
  deleteItem: titulo => {
    dispatch(showModal(titulo, true));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ItemCart);
