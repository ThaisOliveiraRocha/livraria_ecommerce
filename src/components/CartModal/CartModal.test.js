import React from "react";
import { shallow } from "enzyme";
import CartModal from "./CartModal";
import ItemCart from "./ItemCart";

describe("<CartModal />", () => {
  it("should render without crashing", () => {
    shallow(<CartModal />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<CartModal />);
    expect(wrapper).toMatchSnapshot();
  });
});
