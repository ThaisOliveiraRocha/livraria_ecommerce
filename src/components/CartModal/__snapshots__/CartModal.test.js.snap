// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartModal /> should render basic infos 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
